docs/modules/ROOT/pages/how-tos/tls.adoc

```diff
@@ -0,0 +1,116 @@
+= Encrypt database connection
+
+This guide provides an example how to setup a TLS encrypted database connection.
+See the following manuals on how to setup connection encryption:
+
+* https://jdbc.postgresql.org/documentation/head/connect.html#ssl[JDBC connection parameters]
+* https://jdbc.postgresql.org/documentation/head/ssl-client.html[SSL verification methods]
+
+[IMPORTANT]
+====
+The built-in database provider only supports self-signed certificates.
+This is because Keycloak connects through the cluster-internal service and not an externally resolvable resp. verifiable hostname.
+====
+
+====
+Requirements
+
+* `commodore`
+* `kubectl`
+* `openssl`
+* `vault`
+====
+
+. Enable TLS
++
+.Self-signed or custom server certificate (built-in or external database)
+[source,yaml]
+----
+parameters:
+  keycloak:
+    database:
+      tls:
+        enabled: true
+----
++
+.Verified server certificate (only external database)
+[source,yaml]
+----
+parameters:
+  keycloak:
+    database:
+      tls:
+        enabled: true
+        jdbcParams: sslmode=verify-ca&sslrootcert=/etc/ssl/certs/ca-bundle.crt
+        verification: verify
+    # Reset some default parameters:
+        serverCert: ""
+        serverCertKey: ""
+    helm_values:
+      extraVolumes: ""
+      extraVolumeMounts: ""
```

Comment:
This is what I meant by moving the uglyness to the user: Having to reset a bunch of parameters

Reply:
See comments above for some ideas